src/coreclr/CMakeLists.txt

```
@@ -139,7 +139,7 @@ add_subdirectory(tools/aot/jitinterface)
 
 if(NOT CLR_CROSS_COMPONENTS_BUILD)
   # NativeAOT only buildable for a subset of CoreCLR-supported configurations
-  if((CLR_CMAKE_HOST_LINUX OR CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_WIN32) AND (CLR_CMAKE_HOST_ARCH_ARM64 OR CLR_CMAKE_HOST_ARCH_AMD64))
+  if((CLR_CMAKE_HOST_LINUX OR CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_WIN32 OR CLR_CMAKE_HOST_FREEBSD) AND (CLR_CMAKE_HOST_ARCH_ARM64 OR CLR_CMAKE_HOST_ARCH_AMD64))
```

I think at this point we can just make this check the architecture. I'm not aware of any more OSes where CoreCLR builds but NativeAOT doesn't:

```suggestion
  if(CLR_CMAKE_HOST_ARCH_ARM64 OR CLR_CMAKE_HOST_ARCH_AMD64)
```

---

I looked into this a little bit more, the other BSDes have LLVM support so they should be able to build NativeAOT too assuming they can build CoreCLR 

I can't speak for the other community-supported OSes (e.g., haiku, illumos) but changing to just checking arch should not cause issues.